docs/design/design.md

```diff
@@ -0,0 +1,22 @@
+# Design Documentation
+
+This is an attempt to document the OpenShift Installer's existing design principles and also articulate design goals to guide with new feature development and in some cases inform bug fixes.
+
+## Design Goals
+
+### Install methods
+The OpenShift Installer currently supports 2 installation types, UPI (User Provisioned Infrastructure) and IPI (Installer Provisionined Infrastructure). IPI is a highly opinionated method of installing OpenShift. The IPI installation method tries to automate a large portion of the OpenShift Install with the least amount of intervention from the user. While it tries to take care of supporting the most common install scenarios, it cannot be expected to support every install variation that a user wants to use for their cloud.
+
+This is where the UPI offers the greater flexibility required for these highly specialized installs. In this install method, the user is expected to provide and manage several non-OpenShift components needed for the install on their own. For this reason, documentation surrounding the UPI flow are especially important.
+
+The Openshift Installer has the responsibility to validate configuration provided to it. This validation step is common to both the UPI and IPI install methods.
```

Comment:
```suggestion
The OpenShift Installer has the responsibility to validate configuration provided to it. This validation step is common to both the UPI and IPI install methods.
```

I would suggest: 
> The Installer validates user-provided configuration in order to prevent failures during installation.

We do have special validation which is run only in the case of IPI.